docs/language/contracts.rst

```diff
@@ -18,18 +18,21 @@ like so:
       }
   }
 
-A constructor does not have a name and may have any number of arguments. If a constructor has arguments,
-then when the contract is deployed then those arguments must be supplied.
+A constructor can have any number of arguments.
+If a constructor has arguments, those arguments must be supplied when the contract is deployed.
 
 If a contract is expected to hold receive value on instantiation, the constructor should be declared ``payable``.
 
 .. note::
+    In Substrate, constructors have a name. Solang allows to name constructors in the substrate target:
```

Comment:
Not so sure about `In Substrate`. The documentation says `On Substrate` already a lot. We should at least be consistent.